src/app/tests/TestReadInteraction.cpp

```diff
@@ -1744,7 +1739,53 @@ void TestReadInteraction::TestSubscribeUrgentWildcardEvent(nlTestSuite * apSuite
             }
             ctx.GetIOContext().DriveIO(); // at least one IO loop is guaranteed
         }
-        NL_TEST_ASSERT(apSuite, delegate.mGotEventResponse == true);
+        NL_TEST_ASSERT(apSuite, delegate.mGotEventResponse);
```

Comment:
How is this even working today? I'd think that the act of calling `LogEvent` would not actually result in the reporting engine running, so how are we getting events at just past the min interval today?

Reply:
Today, as in before this change, it works because when the min interval timer fires `OnUnblockHoldReportCallback` does:
```
    readHandler->mFlags.Set(ReadHandlerFlags::HoldReport, false);
    if (readHandler->IsDirty())
    {
        InteractionModelEngine::GetInstance()->GetReportingEngine().ScheduleRun();
    }
```
and `LogEvent` ends up marking the handler dirty via `UnblockUrgentEventDelivery` called from `Engine::ScheduleEventDelivery`.

After this change it actually works for basically the same reasons.

Reply:
Ah, makes sense.